src/main/java/com/palantir/gradle/gitversion/GitVersionPlugin.java

```diff
@@ -16,38 +16,56 @@
 
 package com.palantir.gradle.gitversion;
 
+import com.google.common.collect.ImmutableMap;
 import groovy.lang.Closure;
-import java.io.File;
-import java.io.IOException;
-import org.eclipse.jgit.api.Git;
-import org.eclipse.jgit.internal.storage.file.FileRepository;
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
+import org.gradle.api.provider.Provider;
 
 public final class GitVersionPlugin implements Plugin<Project> {
-    private final Timer timer = new Timer();
 
     @Override
     public void apply(final Project project) {
-        project.getRootProject().getPluginManager().apply(GitVersionRootPlugin.class);
+        int maxParallelUsage = 2;
+        Provider<GitVersionCacheService> serviceProvider = project.getGradle()
+                .getSharedServices()
+                .registerIfAbsent("GitVersionCacheService", GitVersionCacheService.class, spec -> {
+                    // Provide some parameters
+                    spec.getMaxParallelUsages().set(maxParallelUsage);
+                });
 
-        final Git git = gitRepo(project);
+        if (project.getRootProject() == project) {
```

Comment:
I think it'd be best to leave this in the root plugin, if possible. Just to separate out the code a bit and avoid this if.

Reply:
I added back the `GitVersionRootPlugin`. The only reason why I deleted the root plugin is that we'd have to have the following in both plugins 
```
        Provider<GitVersionCacheService> serviceProvider = project.getGradle()
                .getSharedServices()
                .registerIfAbsent("GitVersionCacheService", GitVersionCacheService.class, _spec -> {});
```